theano/tensor/elemwise.py

```diff
@@ -1179,7 +1186,7 @@ def c_support_code_apply(self, node, nodename):
         return support_code
 
     def c_code_cache_version_apply(self, node):
-        version = [11]  # the version corresponding to the c code in this Op
+        version = [12]  # the version corresponding to the c code in this Op
```

Comment:
Not very useful, and the consequence are very bad as it will trigger recompilation of ~3000 compiledir for the tests/buildbots. As you have access to the node, can you return 11 if there isn't float16 and return 12 if there if float16?